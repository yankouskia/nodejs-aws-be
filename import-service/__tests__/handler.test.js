const { importProductsFile } = require('../handler');

jest.mock('aws-sdk', () => ({
  S3: jest.fn(() => ({
    getSignedUrl: (_, __, cb) => {
      cb(null, 'presigned-url');
    },
  })),
}));

describe('importProductsFile', () => {
  test('incorrect event returns 500', async () => {
    const result = await importProductsFile({ queryStringParameters: null });

    expect(result.statusCode).toEqual(500);
  });

  test('correct event returns url', async () => {
    const result = await importProductsFile({ queryStringParameters: { name: 'products-file-name' } });

    expect(result.statusCode).toEqual(200);
    expect(result.body).toMatchSnapshot();
  });
});
