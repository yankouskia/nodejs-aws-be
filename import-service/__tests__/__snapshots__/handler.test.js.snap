// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importProductsFile correct event returns url 1`] = `"{\\"uploadURL\\":\\"presigned-url\\"}"`;
